### PR TITLE
Implement typing test

### LINKED ISSUE #52: Implement typing test
**Is your feature request related to a problem? Please describe.**
We are lacking a typing test

**Describe the solution you'd like**
Implement a test using mypy

**Describe alternatives you've considered**
A good approach will be that the test will pass as long as the newly modified files pass mypy.
We can achieve this by selecting the files that have been modified on a later date than when we implement this issue.
Git always keeps the file creation/modify date, so we can make use of it.

We may want to implement a small (bash) script to collect the files that we want to test or adapt this test (at least for now).

**Additional context**
Consider using `annotations` and `typing` in all files. Also consider using the `TYPE_CHECKING` from `typing` to import types only used in annotations:

```python
from __future__ import annotations
from typing import List, TYPE_CHECKING

if TYPE_CHECKING:
  from this.project import CustomType

def check_this_method(foo: List[CustomType]) -> int:
    return len(foo)
```

Mypy documentation: https://mypy.readthedocs.io/en/stable/
Basically:
- Mypy will check all files (checking the imports as well) to see if we are using type annotations
- Mypy will check the typing for all the objects, parameters, return values...

### PR DESCRIPTION
**Pull request by ytturi**
## Description

Close #52

Add a 'checks.py' standalone script in order to implement custom typing tests (find and filter files, then run the tests on those files).

**Observation:**
We can extend this script to execute the other tests on the code as well like black/lint.
We can also run the pytest scripts from it, so it can act as a command handler for the tests.

### CODE REVIEW

File: .github/workflows/pythonapp.yml
Comment (ytturi) [inline .github/workflows/pythonapp.yml:23]: This causes the CI to test the files modified in the current branch, we can update this with a commit hash or a tag instead so it will actually test something for the master branch.

File: checks.py
Comment (fullonic) [inline checks.py:9]: Could we change this to catch all possible `migrations` folder? Maybe something like `./*/migrations`
Because there is also the folder `accounts/migrations` and maybe we could have more on the future. Having a regex path we ensure that all of them will be ignored.

File: .github/workflows/pythonapp.yml
Comment (fullonic) [reply .github/workflows/pythonapp.yml:23]: So for instance, if we add new code to an existent file, `mypy` will check the whole file or only the new code?

**Review by fullonic — CHANGES_REQUESTED** [review]
@francescarpi could we wait for #51 before merge this one? PR #51 formats all the code base and if we merge this PR before, #51 will fail. Thanks

File: checks.py
Comment (ytturi) [reply checks.py:9]: Shouldn't be that hard to parse regex, 👍

File: .github/workflows/pythonapp.yml
Comment (ytturi) [reply .github/workflows/pythonapp.yml:23]: `mypy` will check the whole file.

I would prefer running it against all the code, but for now this should do the trick.

Comment (fullonic) [reply .github/workflows/pythonapp.yml:23]: Ah great. Would be better to solve all the current code base `mypy` errors related with "... no type hints or library stubs" in this PR? Only to avoid these errors on the following PR's.

**Review by XaviTorello — APPROVED** [review]
Looks good, many thanks @ytturi :beers:

File: .github/workflows/pythonapp.yml
Comment (ytturi) [reply .github/workflows/pythonapp.yml:23]: We can fix them when we find them, no need to rush it here in this case I think

Comment (fullonic) [reply .github/workflows/pythonapp.yml:23]: You are right. Will be better do it as we move on :+1:

**Review by fullonic — APPROVED** [review]
:+1:

**Comment by francescarpi** [comment]
Hi! Could you fix conflicts, pls?
Best regards

**Comment by ytturi** [comment]
Done @francescarpi 👍

**Pull request merged**
